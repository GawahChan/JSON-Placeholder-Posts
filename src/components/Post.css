.post {
    border: 1px solid #DDDDDD;
    border-radius: 7px;
    box-shadow: 2px 2px 5px #DDDDDD;
    margin-top: 10px;
    margin-bottom: 10px;
    background-color: #6897C5;
}
.post-title {
    display: flex;
    justify-content: space-between;
    flex: 1;
    border-bottom: 1px solid #DDDDDD;
    padding: 10px;
}

.post-title h3 {
    margin: 0;
    font-weight: 600;
    color: #DDDDDD;
}

.post-body {
    padding: 10px;
    background-color: rgb(245, 245, 245);
}

.btn-showPostBody {
    width: 0; 
    height: 0; 
    border-left: 8px solid transparent;
    border-right: 8px solid transparent;
    cursor: pointer;
}

.btn-showPostBody.up{
    border-top: none;
    border-bottom: 8px solid #DDDDDD;
}

.btn-showPostBody.down {
    border-top: 8px solid #dddddd;
    border-bottom: none;
}