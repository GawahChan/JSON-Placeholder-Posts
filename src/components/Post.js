import React, { Component } from 'react';

class Post extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showPost: false
        }
    }
    togglePosts = () => {
        this.setState({ showPost: !this.state.showPost });
    }

    render() {
        const { userId, postTitle, postBody } = this.props;
        return (
            <div className='post'>
                <div className='post-title'>
                    <h3>{postTitle} #{this.props.postId}</h3>
                    <button className='btn-showPosts' onClick={this.togglePosts}>Show post</button>
                </div>
                {
                    this.state.showPost &&
                    <div className='post-body'>
                        <p>{postBody}</p>
                        <p>Posted by user: {userId}</p>
                    </div>
                }
            </div>
        );
    }
}

export default Post;