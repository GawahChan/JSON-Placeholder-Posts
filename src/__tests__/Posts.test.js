import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Post from '../components/Post';

describe('Posts', () => {
    const postTree = renderer.create(<Post />).toJSON();
    const post = shallow(<Post />);

    it('renders correctly', () => {
        expect(postTree).toMatchSnapshot();
    });

    it('initialises state `showPosts` with false', () => {
        expect(post.state().showPost).toBe(false);
    });

    describe('toggle display for posts', () => {
        beforeEach(() => {
            post.find('.btn-showPosts').simulate('click');
        });
        afterEach(() => {
            post.setState({showPost: false});
        });
        it('sets state `showPost` to true from clicking button once', () => {
            expect(post.state().showPost).toBe(true);
        });
        it('sets state `showPost` to true from clicking button twice', () => {
            expect(post.state().showPost).toBe(true);
            post.find('.btn-showPosts').simulate('click');
            expect(post.state().showPost).toBe(false);
        });
        it('checks if posts in <div className=`post-body` /> is rendered', () => {
            expect(post.find('.post-body').exists()).toBe(true);
            post.find('.btn-showPosts').simulate('click');
            expect(post.find('.post-body').exists()).toBe(false);
        });
    });
})