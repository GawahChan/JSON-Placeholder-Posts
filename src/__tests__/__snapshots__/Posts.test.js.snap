// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts renders correctly 1`] = `
<div
  className="post"
>
  <div
    className="post-title"
  >
    <h3>
       #
    </h3>
    <button
      className="btn-showPosts"
      onClick={[Function]}
    >
      Show post
    </button>
  </div>
</div>
`;
